### ISSUE TITLE
examples/cube: crash when calling Dispose on nil cache

### ISSUE THREAD

**Issue by mewmew** [lakal3/VGE#20]
On rev 64b317fe437fc080eeb73851b8ce7b5954260643, there exist a crash in the cube example, where the `Dispose` method may be called on `nil` cache objects.

```
Using device  Intel(R) HD Graphics 620 (KBL GT2)
Event  {302 660 360 0 30306944}
Event  {103 638 697 0 30306944}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4c96c2]

goroutine 20 [running]:
github.com/lakal3/vge/vge/vk.(*RenderCache).NewFrame(...)
	/home/u/Desktop/play_vge/VGE/vge/vk/rendercache.go:53
github.com/lakal3/vge/vge/vk.(*RenderCache).Dispose(0x0)
	/home/u/Desktop/play_vge/VGE/vge/vk/rendercache.go:27 +0x22
main.(*cubeApp).renderLoop(0x5dc860, 0xc0000be1c0)
	/home/u/Desktop/play_vge/VGE/examples/cube/main.go:186 +0x76
created by main.main
	/home/u/Desktop/play_vge/VGE/examples/cube/main.go:71 +0x265
```

Printf-debugging reveals that cache objects may be `nil` (the slice is allocated ):

```
ca: &{0xc0000cc000 0x5dc860 {[] <nil> map[16:0xc000069260 22:0xc00001a0f8 23:0xc00006e060 24:0xc000014020]} {[] <nil> map[9:0xc000070000 16:0xc000069260 20:0xc000014060 21:0xc000010528]}}
ca: &{0xc0000cc000 0x5dc860 {[] <nil> map[16:0xc0000b9500 22:0xc0000b20c8 23:0xc00009c3f0 24:0xc00009a5c0]} {[] <nil> map[9:0xc0000f2000 16:0xc0000b9500 20:0xc00009a600 21:0xc0000aa570]}}
ca: <nil>
```

Call to printf added to the following line:
```diff
		// If image index < 0, there was some error acquiring image. Most likely window was resized which in many cases mean that we
		// must destroy existing rendering assets like depth buffers etc and recreate them. Next call to GetNextFrame should again succeed
		if imageIndex < 0 {
			// Reset all render caches
			for _, ca := range caches {
+				fmt.Println("ca:", ca)
```

**Comment by mewmew** [comment]
Alternatively to the proposed solution in #21, the `caches` slice handling may be updated such that only non-nil values are stored in `caches` (perhaps making it a map instead of a slice). Currently `caches = make([]*vk.RenderCache, w.GetImageCount())` creates nil cache objects, which may be disposed on the next iteration of the loop, if the `imageIndex` return value of `GetNextFrame` is negative.

**Comment by lakal3** [comment]
Merged solution #21 that should solve the problem in cube example.

Most of VGE uses internally Owner struct to handle ownership of objects. Owner should be resilient to nil values. Owner is not used in cube to make it simple (If Vulkan code can ever be simple).
